nevergrad/optimization/base.py

```diff
@@ -456,6 +456,13 @@ def minimize(
                 if new_sugg:
                     sleeper.start_timer()
             remaining_budget = self.budget - self.num_ask
+            #kills the budget if CMA-ES or FCMA-ES found an optimum
+            #should be replaced by a general concept checking whether an optimization algo signals it wants to stop
+            if hasattr(self, '_es') and (not self._es is None):
```

Comment:
Agreed, that something that we will need (and more generally also a way to specify a stopping criterion for the optimizer.
For now I'm not really fond of adding this specific code here, I'll give it a thought.
@teytaud any idea?

Reply:
What you really would like to do if you have lets say a budget of 100000 and CMA stops after 10000 evals is to have a number of retry. But there is a problem with that: you shouldn't use the same guess, and without bounds you don't know how to generate different guesses.  
If we drop the code above we get wrong time measurements for CMA and FCMA, since
you measure the time for evals after termination of the optimization.

Reply:
For now we've not been investigating much in the actual time needed by the optimizers, since it's we usually assume that this time is negligeable in front of the function evaluation time.
I don't mean it does not make sense (especially given I feel BO optimizers are a pain to benchmark because they are so sloooooow), but we don't have a solution for this so far indeed :s

Reply:
I saw you have benchmarks with very high dimensionality (> 1000). Since covariance matrix adaption has quadratic complexity, the time needed by the optimizer could become relevant. 
Beside that, there are interesting optimization problems like https://www.esa.int/gsp/ACT/projects/gtop/messenger_full/ where evaluation is fast. For this problem fcmaes is four times as fast (including function evaluation) than original cmaes

Reply:
Removed the code from the PR as requested.

Reply:
Actually, if you want to have some mechanism for CMA without without interfering with the base optimizer class, you could change the budget right it the ask or tell method or (F)CMA.
Then again, it's hacky and since there is no framework for this, it may have weird behavior in the benchmarks but we can always remove it if that fails. So far in the benchmarks we have only wanted to compare with the same budget, so some optimizers are asked to start all over again from a different starting point if they consider they have converged.

If you have ideas on how to deal with this for all optimizers correctly please open an issue to discuss it (if you dont have an idea but do want to see the feature, please open it anyway :D) but that may take some time to get it right for the whole package :s